src/Maker/MakeResetPassword.php

```diff
@@ -0,0 +1,277 @@
+        );
+        $input->setArgument(
+            'email-getter',
+            $interactiveSecurityHelper->guessEmailGetter($io, $userClass)
+        );
+        $input->setArgument(
+            'password-setter',
+            $interactiveSecurityHelper->guessPasswordSetter($io, $userClass)
+        );
+    }
+
+    public function generate(
+        InputInterface $input,
+        ConsoleStyle $io,
+        Generator $generator
+    ) {
+        $userClass = $input->getArgument('user-class');
+        $userClassNameDetails = $generator->createClassNameDetails(
+            '\\'.$userClass,
+            'Entity\\'
+        );
+
+        $controllerClassNameDetails = $generator->createClassNameDetails(
+            'ResetPasswordController',
+            'Controller\\'
+        );
+
+        $requestClassNameDetails = $generator->createClassNameDetails(
+            'ResetPasswordRequest',
+            'Entity\\'
+        );
+
+        $repositoryClassNameDetails = $generator->createClassNameDetails(
+            'ResetPasswordRequestRepository',
+            'Repository\\'
+        );
+
+        $requestFormTypeClassNameDetails = $generator->createClassNameDetails(
+            'ResetPasswordRequestFormType',
+            'Form\\'
+        );
+
+        $changePasswordFormTypeClassNameDetails = $generator->createClassNameDetails(
+            'ChangePasswordFormType',
+            'Form\\'
+        );
+
+        $generator->generateController(
+            $controllerClassNameDetails->getFullName(),
+            'resetPassword/ResetPasswordController.tpl.php',
+            [
+                'user_full_class_name' => $userClassNameDetails->getFullName(),
+                'user_class_name' => $userClassNameDetails->getShortName(),
+                'request_form_type_full_class_name' => $requestFormTypeClassNameDetails->getFullName(),
+                'request_form_type_class_name' => $requestFormTypeClassNameDetails->getShortName(),
+                'reset_form_type_full_class_name' => $changePasswordFormTypeClassNameDetails->getFullName(),
+                'reset_form_type_class_name' => $changePasswordFormTypeClassNameDetails->getShortName(),
+                'password_setter' => $input->getArgument('password-setter'),
+                'success_redirect_route' => $input->getArgument('controller-reset-success-redirect'),
+                'from_email' => $input->getArgument('from-email-address'),
+                'from_email_name' => $input->getArgument('from-email-name'),
+                'email_getter' => $input->getArgument('email-getter'),
+            ]
+        );
+
+        $generator->generateClass(
+            $requestClassNameDetails->getFullName(),
+            'resetPassword/ResetPasswordRequest.tpl.php',
+            [
+                'repository_class_name' => $repositoryClassNameDetails->getFullName(),
+                'user_full_class_name' => $userClassNameDetails->getFullName(),
+            ]
+        );
```

Comment:
Let's save this for later (cause I *think* it will take some work), but:

It would be ideal if we could use `EntityClassGenerator` to generate the entity and repository classes. That's our centralized entry point for creating these classes and making sure they all look uniform. We would use that class and *then* use the `ClassSourceManipulator` to add in our custom code (that's the tricky part - writing PHP Parser code to add methods, etc take a bit of work to get right).

Reply:
made a note on the board to implement in the near future..